File: apps/motoboy-pwa/src/deliverySections.ts
import type { Delivery } from "./types";

const AVAILABLE_DELIVERY_STATUS: Delivery["status"] = "AGUARDANDO_MOTOBOY";
const ACTIVE_DELIVERY_STATUSES = new Set<Delivery["status"]>([
  "ACEITA_PELO_MOTOBOY",
  "COLETADA",
  "EM_ROTA",
  "PROBLEMA",
]);

export type DeliverySections = {
  active: Delivery[];
  available: Delivery[];
};

/** Separa a fila do motoboy entre entregas disponiveis e entregas que ja estao sob atendimento. */
export function deliverySections(deliveries: Delivery[]): DeliverySections {
  return {
    active: deliveries.filter((delivery) => ACTIVE_DELIVERY_STATUSES.has(delivery.status)),
    available: deliveries.filter((delivery) => delivery.status === AVAILABLE_DELIVERY_STATUS),
  };
}

File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliverySections.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import com.drogsantoantonio.farmadelivery.data.models.DeliveryDto

data class DeliverySections(
  val active: List<DeliveryDto>,
  val available: List<DeliveryDto>,
)

fun deliverySections(deliveries: List<DeliveryDto>): DeliverySections {
  return DeliverySections(
    active = deliveries.filter { it.status in ACTIVE_DELIVERY_STATUSES },
    available = deliveries.filter { it.status == AVAILABLE_DELIVERY_STATUS },
  )
}

private val ACTIVE_DELIVERY_STATUSES = setOf(
  "ACEITA_PELO_MOTOBOY",
  "COLETADA",
  "EM_ROTA",
  "PROBLEMA",
)

private const val AVAILABLE_DELIVERY_STATUS = "AGUARDANDO_MOTOBOY"
